File: PopupQToolButton.py
# -*- coding: utf-8 -*-

from PyQt5.QtCore import Qt
from PyQt5.QtWidgets import QToolButton

class PopupQToolButton(QToolButton):
    def __init__(self, parent=None):
        QToolButton.__init__(self, parent)
        self.setPopupMode(QToolButton.MenuButtonPopup)
        self.setToolButtonStyle(Qt.ToolButtonTextUnderIcon)
        self.triggered.connect(self.setDefaultAction)

File: PopupQToolButton.cpp
#include "PopupQToolButton.h"
#include <QAction>

PopupQToolButton::PopupQToolButton(QWidget * parent) : QToolButton(parent)
{
    this->setPopupMode(QToolButton::MenuButtonPopup);
    this->setToolButtonStyle(Qt::ToolButtonTextUnderIcon);
    QObject::connect(this, &QToolButton::triggered, this, &QToolButton::setDefaultAction);
}

PopupQToolButton::~PopupQToolButton()
{
}
